Keywords/commonUtils/WebTableUtils.groovy

```diff
@@ -1,253 +1,285 @@
+				count=0;
+				List<WebElement> cells = rows.get(i).findElements(By.tagName("td"));
+				for (WebElement cell : cells) {
+					// Extract data from each cell and add it to the list
+					if (count == 0) {
+						rowData = cell.getText();
+					} else {
+						rowData = rowData + "," + cell.getText();
+					}
+					count++;
+				}
+				tableData.add(rowData);
+			}
+		}
+		catch (Exception e) {
+			WebUI.comment("Exception :: " + e.getMessage())
+			KeywordUtil.markFailedAndStop("Exception while reading data from WebTable");
+		}
+		return tableData;
+	}
+
+	@Keyword(keywordObject = "WebTable")
+	/**
+	 * Get Web table cell value
+	 * @param webTableObject - WebTable object referenced from OR
+	 * @param primaryKey - primary key of WebTable
+	 * @param primaryKeyValue - primary key value of row
+	 * @param columnName  - column to fetch cell value
+	 */
+	public static String getWebTableCellValue(TestObject webTableObject,String primaryKey,String rowName,String columnName) {
+		try {
+			String cellValue;
+			Map<String, Map<String, String>> webTableMap = getWebTableAsMap(webTableObject, primaryKey);
+			//Iterating through Webtable maps
+			for (Entry<String, Map<String, String>> entry : webTableMap.entrySet()) {
+				if (entry.getKey().equals(rowName)) {
+					cellValue = entry.getValue().get(columnName);
+					break; //Exit inner loop once found
+				}
+			}
+			if (cellValue != null) {
+				KeywordUtil.markPassed("Cell Value for row name: '" + rowName + "' and column name: '" + columnName+"' is '"+cellValue+"'")
+			} else {
+				KeywordUtil.markFailedAndStop("Cell with row name '" + rowName + "' and column name '" + columnName + "' not found in the table with primarykey: '"+primaryKey+"'")
+			}
+			return cellValue;
+		}
+		catch (Exception e) {
+			WebUI.comment("Exception :: " + e.getMessage())
+			KeywordUtil.markFailedAndStop("Exception while fetching cell value from WebTable");
+		}
+	}
+	/**
+	 * to get the read only cells
+	 * @param tableTitle - Title of the table to locate
+	 * @param disabled   - The class that indicates a readOnly state
+	 *
+	 */
+	@Keyword(keywordObject = 'readOnly')
+	public static void verifyTableCellIsEnabled(String tableTitle, String disabled) {
+		try {
+			WebElement table = WebUiBuiltInKeywords.findWebElement(new TestObject().addProperty("title", ConditionType.EQUALS, tableTitle), 30);
+			// Iterate through rows
+			for (WebElement row : table.findElements(By.tagName("tr"))) {
+				List<WebElement> allCells = row.findElements(By.tagName("td"));
+				// Iterate through cells in each row
+				for (int i = 0; i < allCells.size(); i++) {
+					WebElement cell = allCells.get(i);
+					String cellClass = cell.getAttribute("class");
+					if (cellClass != null) {
+						if (cellClass.equals("disabled")) {
+							KeywordUtil.logInfo("Cell is in readonly mode with value " + cell.getText());
+						} else {
+							KeywordUtil.markFailedAndStop("Cell is enabled mode with value " + cell.getText());
+						}

```

Comment:
No need to print cell value for pass condition. If there are 100 cells, report will become unreadable
Just log failure and stop if condition is not met.

Reply:
Done